cli/central-client/src/main/java/org/ballerinalang/central/client/CentralAPIClient.java

```diff
@@ -0,0 +1,380 @@
+            String supportedPlatform, boolean isBuild) {
+        LogFormatter logFormatter = new LogFormatter();
+        if (isBuild) {
+            logFormatter = new BuildLogFormatter();
+        }
+
+        String url = PACKAGES + "/" + org + "/" + name;
+        // append version to url if available
+        if (null != version && !version.isEmpty()) {
+            url = url + "/" + version;
+        }
+
+        initializeSsl();
+        HttpURLConnection conn = createHttpUrlConnection(url);
+        conn.setInstanceFollowRedirects(false);
+        setRequestMethod(conn, Utils.RequestMethod.GET);
+
+        // Set headers
+        conn.setRequestProperty(BALLERINA_PLATFORM, supportedPlatform);
+        conn.setRequestProperty(BAL_LANG_SPEC_VERSION, IMPLEMENTATION_VERSION);
+        conn.setRequestProperty(ACCEPT_ENCODING, IDENTITY);
+        conn.setRequestProperty(USER_AGENT, RepoUtils.getBallerinaVersion());
+
+        try {
+            // 302   - Package is found
+            // Other - Error occurred, json returned with the error message
+            if (getStatusCode(conn) == HttpURLConnection.HTTP_MOVED_TEMP) {
+                // get redirect url from "location" header field
+                String newUrl = conn.getHeaderField(LOCATION);
+                String contentDisposition = conn.getHeaderField(CONTENT_DISPOSITION);
+
+                conn = createHttpUrlConnection(newUrl);
+                conn.setRequestProperty(CONTENT_DISPOSITION, contentDisposition);
+
+                boolean isNightlyBuild = RepoUtils.getBallerinaVersion().contains("SNAPSHOT");
+                createBaloInHomeRepo(conn, packagePathInBaloCache, org + "/" + name, isNightlyBuild, newUrl,
+                        contentDisposition, outStream, logFormatter);
+            } else {
+                try (BufferedReader reader = new BufferedReader(
+                        new InputStreamReader(conn.getErrorStream(), Charset.defaultCharset()))) {
+                    Error errorJsonSchema = new Gson().fromJson(reader, Error.class);
+                    throw ErrorUtil
+                            .createCommandException(logFormatter.formatLog("error: " + errorJsonSchema.getMessage()));
+                } catch (IOException e) {
+                    throw ErrorUtil.createCommandException(logFormatter.formatLog(
+                            "failed to pull the package '" + org + "/" + name + "' from the remote repository '" + url
+                                    + "'"));
+                }
+            }
+        } catch (Exception e) {
+            throw ErrorUtil.createCommandException(e.getMessage());
+        } finally {
+            conn.disconnect();
+            Authenticator.setDefault(null);
+        }
+    }
+
+    /**
+     * Search packages in registry
+     */
+    public PackageSearchResult searchPackage(String query) {
+        initializeSsl();
+        HttpURLConnection conn = createHttpUrlConnection(PACKAGES + "/" + "?q=" + query);
```

Comment:
`"/" + "?q="` this can be a single string.

Reply:
Done